src/backend/elfobj.c

```diff
@@ -1689,11 +1691,66 @@ STATIC void setup_comdat(Symbol *s)
     symbol_debug(s);
     if (tyfunc(s->ty()))
     {
+        reset_symbuf->write(&s, sizeof(s));
+
         //s->Sfl = FLcode;      // was FLoncecode
-        //prefix = ".gnu.linkonce.t";   // doesn't work, despite documentation
-        prefix = ".text.";              // undocumented, but works
-        type = SHT_PROGBITS;
-        flags = SHF_ALLOC|SHF_EXECINSTR;
+        /* ".gnu.linkonce.t" sections result in "Too many sections" errors from ld.
+         * prefix = ".gnu.linkonce.t";
+         */
+
+        /* Get section index of the comdat
+         */
+        IDXSEC comdatidx = section_cnt + 1;
+
+        /* create a weak symbol with the group signature in it.
+         *    4  .22 .weak _Z3barIiEiT_,@FUNCTION,VALUE=.text._Z3barIiEiT_+0x00,SIZE=14
+         */
+        const char *p = cpp_mangle(s);
+        IDXSTR namidx = Obj::addstr(symtab_strings, p);
+        IDXSYM info = elf_addsym(namidx, 0, 0, STT_FUNC, STB_WEAK, comdatidx);
```

Comment:
I'm curious why you're getting that error. I'm getting a different one, from both the autotester and my setup at home, where cppa in the test suite is crashing. I spent several hours on that last night and am still not able to determine the cause. It's maddening.